osvimdriver/openstack/neutron/driver.py

```diff
@@ -14,36 +18,69 @@ def __init__(self, session):
+            common._generate_additional_logs(e, 'received', external_request_id, 'plain/text',
+                                       'response', 'http', {'status_code' : e.status_code,'status_reason_phrase' : e.message }, driver_request_id)
+            raise e
 
-    def get_network_by_name(self, network_name):
+
+    def get_network_by_name(self, network_name,driver_request_id=None):
         if network_name is None:
             raise ValueError('network_name must be provided')
         neutron_client = self.__get_neutron_client()
         logger.debug('Retrieving network with name %s', network_name)
-        result = neutron_client.list_networks()
+        external_request_id = str(uuid.uuid4())
+        common._generate_additional_logs('', 'sent', external_request_id, '',
+                                       'request', 'http', {'method' : 'get', 'uri' : LOG_URI_PREFIX +'/networks' }, driver_request_id)
+        try:
+            result = neutron_client.list_networks()
+            common._generate_additional_logs(str(result), 'received', external_request_id, 'plain/text',
+                                       'response', 'http', {'status_code' : 200, 'status_reason_phrase' : 'ok'}, driver_request_id)
+        except Exception as e:
+            common._generate_additional_logs(e, 'received', external_request_id, 'plain/text',
+                                       'response', 'http', {'status_code' : e.status_code,'status_reason_phrase' : e.message }, driver_request_id)
+            raise e    
         matches = []
         for network in result['networks']:
             if network['name'] == network_name:
                 matches.append(network)
         if len(matches) > 1:
             raise neutronexceptions.NeutronClientNoUniqueMatch(resource='Network',
                                                                name=network_name)
-        elif len(matches) == 1:
+        elif len(matches) == 1:  
             return matches[0]
         else:
             raise neutronexceptions.NotFound(message='Unable to find network with name \'{0}\''.format(network_name))
 
-    def get_subnet_by_id(self, subnet_id):
+    def get_subnet_by_id(self, subnet_id,driver_request_id=None):
         if subnet_id is None:
             raise ValueError('subnet_id must be provided')
         neutron_client = self.__get_neutron_client()
         logger.debug('Retrieving subnet with id %s', subnet_id)
-        result = neutron_client.show_subnet(subnet_id)
-        return result['subnet']
+        try:
+            external_request_id = str(uuid.uuid4())
+            common._generate_additional_logs('', 'sent', external_request_id, '',
+                                       'request', 'http', {'method' : 'get', 'uri' : LOG_URI_PREFIX +'/subnets/' + subnet_id}, driver_request_id)
+            result = neutron_client.show_subnet(subnet_id)
+            common._generate_additional_logs(result, 'received', external_request_id, 'plain/text',
```

Comment:
I am expecting the library would return the values in dict/json. May be you can update the content type to application/json and convert the result to str.

Reply:
I have made changes.